core/src/main/java/org/apache/druid/java/util/common/StringUtils.java

```diff
@@ -399,7 +399,13 @@ public static String repeat(String s, int count)
       return new String(single, StandardCharsets.UTF_8);
     }
     if (Integer.MAX_VALUE / count < len) {
-      throw new OutOfMemoryError();
+      String errMsg = StringUtils.format(
```

Comment:
I would go for a concise one `The produced string is too large.` I think a general Druid user is smart enough to understand it right away.

Reply:
@egor-ryashin Done.